### PR TITLE
Implemented query params for search

### PR DESCRIPTION
**Pull request by Kachyr**
#213  Task
Refactored routing, searchField, and improvements for searchResults

### CODE REVIEW

File: src/components/SearchField.tsx [hunk 1]
```diff
@@ -46,10 +48,11 @@ const SearchField: React.FC = () => {
   };
 
   const submitSearchRequest = (event: React.SyntheticEvent) => {
+    setDropdown(false);
     event.preventDefault();
     if (inputValue.trim()) {
       history.push(
-        `/search-results/${inputValue}/${genreToSearch}/${dateRange.fromYear}/${dateRange.toYear}/${checkboxes.favorites}/${checkboxes.history}/1`
+        `/search-results?title=${inputValue}&genre=${genreToSearch}&fromYear=${dateRange.fromYear}&toYear=${dateRange.toYear}&favorites=${checkboxes.favorites}&history=${checkboxes.history}&page=1`
```
Comment (oleksandrstarov) [inline src/components/SearchField.tsx:55]: Please use search param for query, and create query via URLSearchParams().
Also avoid setting undefined values there - it doesent looks nice

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/components/SearchField.tsx (hunk 1 above)
Comment (Kachyr) [reply src/components/SearchField.tsx]: I've refactored it but don't know if it's good.

File: src/components/SearchField.tsx [hunk 2]
```diff
@@ -45,12 +45,24 @@ const SearchField: React.FC = () => {
...
+    const queryString = new URLSearchParams({
+      title: inputValue,
+      genre: genreToSearch,
+      fromYear: dateRange.fromYear.toString(),
+      toYear: dateRange.toYear.toString(),
+      favorites: checkboxes.favorites.toString(),
+      history: checkboxes.history.toString(),
+      page: '1'
+    });
+
```
Comment (oleksandrstarov) [inline src/components/SearchField.tsx:60]: Nice!

**Review by oleksandrstarov — APPROVED** [review]
LGTM

**Review by solomiia-sahala — APPROVED** [review]
LGTM

**Pull request merged**